src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -35,10 +44,9 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
 
             GeoServerResourceLoader loader =
                     GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            File f =
-                    Resources.find(
-                            Resources.fromURL(Files.asResource(loader.getBaseDirectory()), path),
-                            true);
+            Resource resource = loader.get(Paths.convert(path));
```

Comment:
You don't want to involve `File` at all, but work purely based on `Resource`. Looking at `GeoServerResourceLoader.fromURL(URL url) -> Resources.fromURL(Resource baseDirectory, URL url)`, all you need to do is return `new URL("resource://" + path)`, provided that `path` is relative to the root resource.

Reply:
geotools doesn't support resource as a protocol. As a consequence it creates an exception when trying to create the datastore

Reply:
For the moment it is not possible to avoid creating a File without changes in geotools

Reply:
@jodygarnett I have no plan for the moment to dig into geotools.

Reply:
@groldan I think your change has been resolved? Perhaps ...

The use of ``resource.file()`` does unpack the file to the local system, but that is later ignored and ``resource.in()`` is used for direct file / blob access.

I think the code is much cleaner